### ISSUE TITLE
More than 1000 expressions in a list are not allowed on Oracle

### ISSUE THREAD

**Issue by solracsf** [nextcloud/activity#1384]
```
{
  "reqId": "LTByxz66jOYmmphsrxpM",
  "level": 3,
  "time": "2023-10-28T03:54:00+02:00",
  "remoteAddr": "",
  "user": "--",
  "app": "core",
  "method": "",
  "url": "--",
  "message": "More than 1000 expressions in a list are not allowed on Oracle.",
  "userAgent": "--",
  "version": "26.0.8.2",
  "exception": {
    "Exception": "Doctrine\\DBAL\\Query\\QueryException",
    "Message": "More than 1000 expressions in a list are not allowed on Oracle.",
    "Code": 0,
    "Trace": [
      {
        "file": "/var/www/nextcloud/lib/private/DB/QueryBuilder/QueryBuilder.php",
        "line": 326,
        "function": "execute",
        "class": "OC\\DB\\QueryBuilder\\QueryBuilder",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/activity/lib/Data.php",
        "line": 513,
        "function": "executeStatement",
        "class": "OC\\DB\\QueryBuilder\\QueryBuilder",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/activity/lib/Data.php",
        "line": 380,
        "function": "deleteActivitiesForMySQL",
        "class": "OCA\\Activity\\Data",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/activity/lib/Data.php",
        "line": 365,
        "function": "deleteActivities",
        "class": "OCA\\Activity\\Data",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/activity/lib/BackgroundJob/ExpireActivities.php",
        "line": 54,
        "function": "expire",
        "class": "OCA\\Activity\\Data",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/public/BackgroundJob/Job.php",
        "line": 78,
        "function": "run",
        "class": "OCA\\Activity\\BackgroundJob\\ExpireActivities",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/public/BackgroundJob/TimedJob.php",
        "line": 103,
        "function": "start",
        "class": "OCP\\BackgroundJob\\Job",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/public/BackgroundJob/TimedJob.php",
        "line": 93,
        "function": "start",
        "class": "OCP\\BackgroundJob\\TimedJob",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/cron.php",
        "line": 152,
        "function": "execute",
        "class": "OCP\\BackgroundJob\\TimedJob",
        "type": "->"
      }
    ],
    "File": "/var/www/nextcloud/lib/private/DB/QueryBuilder/QueryBuilder.php",
    "Line": 263,
    "message": "More than 1000 expressions in a list are not allowed on Oracle.",
    "query": "DELETE FROM `*PREFIX*activity` WHERE `activity_id` IN (:ids)",
    "exception": [],
    "CustomMessage": "More than 1000 expressions in a list are not allowed on Oracle."
  },
  "id": "653cefa406c8b"
}
```

**Comment by solracsf** [comment]
@miaulalala shouldn't https://github.com/nextcloud/activity/pull/1359 fix this (v26.0.8 here)?
If yes, can it be backported?

**Comment by miaulalala** [comment]
Yes it should be backported but I forgot to do it since the backportbot doesn't like it :sob: 

Will do so manually this week.
